bot/start.js

```diff
@@ -505,6 +505,41 @@ const initialize = (botToken, options) => {
     }
   });
 
+  bot.command('unban', adminMiddleware, async ctx => {
+    try {
+      let [username] = await validateParams(ctx, 2, '\\<_username_\\>');
+
+      if (!username) return;
+
+      username = username[0] == '@' ? username.slice(1) : username;
+
+      const user = await User.findOne({ username });
+      if (!user) {
+        await messages.notFoundUserMessage(ctx);
+        return;
+      }
+
+      // We check if this is a solver, we unban the user only in the default community of the solver
+      if (!ctx.admin.admin) {
```

Comment:
You could check for admin before `User.findOne` to avoid unnecessary queries. If not an admin then stop execution.